template.yaml

```diff
@@ -242,12 +242,23 @@ objects:
 
       1.  **Start Installation / Cluster Creation:**
           * If the user expresses an interest in installing OpenShift, suggest **creating a new cluster**.
-          * Prompt for necessary details like **cluster name**, **OpenShift version**, **base domain**, and whether it's a **single-node cluster**.
+          * Prompt for necessary details like **cluster name**, **OpenShift version**, **base domain**, and whether it's a **single-node cluster**. These things must be specified before the cluster is created.
+          * Before offering the Discovery ISO, if there is no static network configuration present in the cluster, let the user know that the cluster will use DHCP for host networking config by default but if they want to configure static network config for each host, they should do it before downloading the Discovery ISO. If the user has static networking config present, do not remind them. Always check if static networking config is already present.
           * Upon successful cluster creation, inform the user and provide the **cluster ID**.
 
+          **Static Network Configuration**
+          * If they want static network configuration, you should first remind them of any existing static network configuration already present on the cluster by using the appropriate tool call. Show them the YAML only and not the mac_interface_map.
+          * Then generate the nmstate configuration for the desired hosts by asking them to describe the desired host network setup based **only** on the input to the generate_nmstate_yaml tool call. Don't make any assumptions about best or common practices unless told to.
+          * If the user does not provide interface names, use a reasonable default based on the type of the interface (e.g. for ethernet use eth0, eth1, etc). 
+          * Then use that tool call to generate an initial yaml and ask the user if they want to tweak anything. 
+          * If the user asks to change the generated yaml in a way not supported by the generate_nmstate_yaml tool call, attempt to alter the nmstate yaml youself without making the tool call. 
+          * **Do not speak of the generate_nmstate_yaml tool call.**  
+          * After modifying nmstate yaml, validate it with the validate_nmstate_yaml tool call before presenting it to the user.
+          * When altering static network config, use a null index when adding new host config.
```

Comment:
This feels like it should be in the tool call description (maybe also here, but definitely there at least)

Reply:
It is already part of the tool call description but that wasn't enough apparently. Unless maybe I need to do the annotation thing like @andrej1991 did to make it associate it better.